periodictable/nsf.py

```diff
@@ -466,32 +472,57 @@ def scattering(self, wavelength=ABSORPTION_WAVELENGTH):
         if not self.has_sld():
             return None, None, None
 
-        b_c = self.b_c
-        sigma_s = self.total
-        sigma_a = self.absorption*wavelength/ABSORPTION_WAVELENGTH
-
         number_density = self._number_density*1e-24
+        b_c, b_pp, sigma_s = self.xs_by_wavelength(wavelength)
+        return _calculate_scattering(number_density, wavelength, b_c, b_pp, sigma_s)
 
-        # PAK 2017-04-21: compute incoherent xs from total xs
-        sigma_c = 4*pi/100 * b_c**2
-        sigma_i = max(sigma_s - sigma_c, 0.)
+def _calculate_scattering(number_density, wavelength, b_c, b_pp, sigma_s):
+    #print("in scat", number_density, wavelength, b_c, b_pp, sigma_s)
+    # Compute SLD
+    sld_re = 10*number_density * b_c
```

Comment:
Can you put a comment in here that indicates the meaning of the constant +10? (and -10 on the next line)

Reply:
I put more details about units at every step of the calculation.

This would be an excellent use for macros: we want to make sure that the units are are correct on the calculations but only at compile time. We don't want to check every calculation at runtime. So for now we just need to be careful.

With a really rich type system we could set it up so that the linter catches errors.